gobblin-utility/src/main/java/gobblin/util/concurrent/TaskSchedulerFactory.java

```diff
@@ -0,0 +1,113 @@
+    SCHEDULEDEXECUTORSERVICE(ScheduledExecutorServiceTaskScheduler.class),
+    HASHEDWHEELTIMER(HashedWheelTimerTaskScheduler.class);
+
+    @Getter
+    private final Class<? extends TaskScheduler> taskSchedulerClass;
+
+    public static Optional<TaskSchedulerType> parse(Optional<String> name) {
+      if (!name.isPresent() || StringUtils.isEmpty(name.get())) {
+        return Optional.absent();
+      }
+      return Enums.getIfPresent(TaskSchedulerType.class, name.get().toUpperCase());
+    }
+  }
+
+  private TaskSchedulerFactory() {
+  }
+
+  /**
+   * Gets an instance of the default {@link TaskScheduler} and ensures that it is started.
+   *
+   * @param <K> the type of the key for the {@link ScheduledTask}
+   * @param <T> the type of the {@link ScheduledTask}
+   * @return an instance of {@link TaskScheduler}
+   */
+  public static <K, T extends ScheduledTask<K>> TaskScheduler<K, T> get() {
+    return getImpl(Optional.<String>absent(), Optional.<String>absent());
+  }
+
+  /**
+   * Gets an instance of the default {@link TaskScheduler} and ensures that it is started.
+   *
+   * @param name the name associated threads created by the {@link TaskScheduler}
+   * @param <K> the type of the key for the {@link ScheduledTask}
+   * @param <T> the type of the {@link ScheduledTask}
+   * @return an instance of {@link TaskScheduler}
+   */
+  public static <K, T extends ScheduledTask<K>> TaskScheduler<K, T> get(Optional<String> name) {
+    return getImpl(Optional.<String>absent(), name);
+  }
+
+  /**
+   * Gets an instance of the {@link TaskScheduler} with the specified type and ensures that it is started. If
+   * the type is unknown an instance of the default {@link TaskScheduler} will be returned.
+   *
+   * @param type the type of the {@link TaskScheduler}
+   * @param <K> the type of the key for the {@link ScheduledTask}
+   * @param <T> the type of the {@link ScheduledTask}
+   * @return an instance of {@link TaskScheduler}
+   */
+  public static <K, T extends ScheduledTask<K>> TaskScheduler<K, T> get(String type) {
+    return getImpl(Optional.fromNullable(type), Optional.<String>absent());
+  }
+
+  /**
+   * Gets an instance of the {@link TaskScheduler} with the specified type and ensures that it is started. If
+   * the type is unknown an instance of the default {@link TaskScheduler} will be returned.
+   *
+   * @param type the type of the {@link TaskScheduler}
+   * @param name the name associated threads created by the {@link TaskScheduler}
+   * @param <K> the type of the key for the {@link ScheduledTask}
+   * @param <T> the type of the {@link ScheduledTask}
+   * @return an instance of {@link TaskScheduler}
+   */
+  public static <K, T extends ScheduledTask<K>> TaskScheduler<K, T> get(String type, Optional<String> name) {
+    return getImpl(Optional.fromNullable(type), name);
+  }
+
+  private static <K, T extends ScheduledTask<K>> TaskScheduler<K, T> getImpl(Optional<String> type, Optional<String> name) {
```

Comment:
Should this signature be in fact `getImpl(String, Optional<String>)`? Given that type is not actually optional? The methods with no type can get the default type themselves.